homeassistant/config_entries.py

```diff
@@ -1858,6 +1858,35 @@ class OptionsFlow(data_entry_flow.FlowHandler):
 
     handler: str
 
+    @callback
+    def _async_abort_entries_match(
+        self, match_dict: dict[str, Any] | None = None
+    ) -> dict[str, str]:
+        """Validate the user input against other config entries."""
+        if match_dict is None:
+            match_dict = {}  # Match any entry
+
+        config_entry = cast(
+            ConfigEntry, self.hass.config_entries.async_get_entry(self.handler)
+        )
+        errors: dict[str, str] = {}
+        for entry in [
+            entry
+            for entry in self.hass.config_entries.async_entries(config_entry.domain)
+            if entry is not config_entry and entry.source != SOURCE_IGNORE
+        ]:
+            if all(
+                item
+                in ChainMap(
+                    entry.options,  # type: ignore[arg-type]
+                    entry.data,  # type: ignore[arg-type]
+                ).items()
+                for item in match_dict.items()
+            ):
+                errors["base"] = "already_configured"
+                break
+        return errors
```

Comment:
We should raise AbortFlow as done in the config flow. The integration can catch this if wanted and eg show an error instead.